pkg/operator/v1helpers/helpers.go

```diff
@@ -14,13 +14,44 @@ import (
 	"k8s.io/apimachinery/pkg/runtime"
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 	"k8s.io/client-go/util/retry"
+	"k8s.io/klog/v2"
+	"k8s.io/utils/diff"
 
 	"github.com/ghodss/yaml"
 
 	configv1 "github.com/openshift/api/config/v1"
 	operatorv1 "github.com/openshift/api/operator/v1"
 )
 
+var _ OperatorStatusUpdater = &LoggingStatusUpdater{}
+
+type LoggingStatusUpdater struct {
+	// cachedConditions store the last observed conditions to be returned in the
+	// GetOperatorState method in order to prevent excessive conditions update logging
+	cachedConditions []operatorv1.OperatorCondition
```

Comment:
we don't have to provide any synchronization for reads/writes? Here, `GetOperatorState` and `UpdateOperatorStatus` must not be used concurrently. We should at least make this clear in the constructor.

Reply:
you're right, we should